pkg/internal/codegen/parse/index.go

```diff
@@ -184,28 +183,38 @@ type resourceTags struct {
 }
 
 // ParseResourceTag parses the tags in a "+resource=" comment into a resourceTags struct
-func parseResourceTag(tag string) resourceTags {
+func parseResourceTag(t *types.Type) (resourceTags, error) {
```

Comment:
I'd keep this as two separate functions -- it improves the testability and makes it easier to reuse.